### PR TITLE
test: improve testing (DEV-152)

### PR DESCRIPTION
**Pull request by irinaschubert**
resolves DEV-152

### CODE REVIEW

File: test/e2e/test_connection.py [hunk 1]
```diff
@@ -1,37 +1,43 @@
...
-        self.assertRaises(BaseError, con.get, "/gagaga")
+        self.assertRaises(BaseError, self.con.get, "/doesNotExist")
+        self.con.logout()
+        self.assertIsNotNone(res['@graph'])
+        self.assertRaises(BaseError, self.con.get, "/doesNotExist")
 
-    def test_put(self):
+    def test_put(self) -> None:
+        # TODO add test
         pass
```
Comment (BalduinLandolt) [inline test/e2e/test_connection.py:37]: that's my style of test! worthy of dsp-metadata-gui! xD

File: test/e2e/test_connection.py [hunk 2]
```diff
@@ -51,15 +57,16 @@ def test_post(self):
...
         self.assertIsNotNone(res['@id'])
-        id = res['@id']
         self.assertEqual(res['@type'], 'anything:Thing')
         self.assertEqual(res['rdfs:label'], 'knora-py thing')
 
-        eraseinfo = """{
+        res_id = res['@id']
+
+        erase_info = """{
             "@id" : "%s",
```
Comment (BalduinLandolt) [inline test/e2e/test_connection.py:70]: personally, I have a huge dislike for these old school C-style formatted strings.  I much prefer `f"...{id}"`

**Review by BalduinLandolt — APPROVED** [review]
Looks good all in all.

I noted some things that struck my eyes, but that's all very minor.

Finally, some of the tests have changed so much that combined with moving the files, github wouldn't show me a diff for them. In these cases I couldn't review quite as thorroughly. But on a quick glance they looked good.

File: knora/dsplib/models/user.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import json
 import os
 import sys
+import time
```
[the remark this answers is not in the document]
Comment (irinaschubert) [reply knora/dsplib/models/user.py:4]: I didn't touch these files but I reformatted it now.

File: knora/dsplib/utils/expand_all_lists.py [hunk 4]
```diff
@@ -19,11 +19,12 @@ def expand_lists_from_excel(data_model: Dict) -> List[str]:
     if lists is not None:
```
[the remark this answers is not in the document]
Comment (irinaschubert) [reply knora/dsplib/utils/expand_all_lists.py:19]: good point, used `if not lists` instead

File: test/e2e/test_connection.py (hunk 2 above)
Comment (irinaschubert) [reply test/e2e/test_connection.py]: replaced it ;)

**Review by BalduinLandolt — APPROVED** [review]
looks good

**Pull request merged**